icaruscode/CRT/CRTUtils/CRTHitRecoAlg.cc

```diff
@@ -393,27 +392,41 @@ sbn::crt::CRTHit CRTHitRecoAlg::MakeTopHit(art::Ptr<CRTData> data, ULong64_t Glo
     auto const& adsGeo = adGeo.SensitiveVolume(adsid_max); //trigger strip
     uint64_t thit = data->fTs0;
     Long64_t thit1 = data->fTs1;
-
-    if(adsid_max<8){
-        thit -= (uint64_t)round(abs((92+hitpos.X())*fPropDelay));
-        thit1 -= (uint64_t)round(abs((92+hitpos.X())*fPropDelay));
+    thit -= fSiPMtoFEBdelay;
+    thit1 -= fSiPMtoFEBdelay;
+    //thit1 = (Long64_t)(thit-GlobalTrigger[(int)mac+73]);
+    //thit -= (uint64_t)round(abs((92+hitpos.Z())*fPropDelay));
+    //std::cout<<"Hit position local: X "<<hitpos.X()<<" Z "<<hitpos.Z();
+    if((92-hitpos.X())<=(92+hitpos.Z())) {
```

Comment:
It would be nice to have a comment explaining this magic.

Reply:
Sure! I will include a comment in the code:
basically the magic number 92 comes from the fact that the bars are 184 cm long, 92 cm is the center.
center of the module is located at coordinates (0,0), this means that when you compute the distance of the hit from the SiPM, you have to translate it by 92 cm.
Concerning why for X is (92-hitpos.X) while for Z is (92+hitpos.Z) this is due to a bug/feature of the geometry, the local position of the modules are defined before rotation operations, due to some weird effect, it is like the readout of the bar happens to be on -92cm in X in local coordinates and then is located correctly after module placement. Anyway, light corrections are applied to local values, not to reconstructed CRT hit positions, therefore this fixes it. Z coordinates was fine.